pallets/corporate-actions/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+        ///
+        /// To understand withholding tax, for example, let's assume that you hold IKEA shares.
+        /// IKEA now decides to distribute 100 SEK to Alice.
+        /// Alice lives in Sweden, so Skatteverket (the Swedish tax authority) wants 30% of that.
+        /// Then those 100 * 30% are withheld from Alice, and IKEA will send them to Skatteverket.
+        ///
+        /// (ticker => % to withhold)
+        pub DefaultWitholdingTax get(fn default_withholding_tax): map hasher(blake2_128_concat) Ticker => Permill;
+
+        /// The amount of tax to withhold ("withholding tax", WT) for a certain ticker x DID.
+        /// If an entry exists for a certain DID, it overrides the default in `DefaultWithholdingTax`.
+        ///
+        /// (ticker => DID => % to withhold)
+        pub DidWitholdingTax get(fn did_withholding_tax):
+            double_map hasher(blake2_128_concat) Ticker, hasher(blake2_128_concat) IdentityId => Option<Permill>;
+    }
+}
+
+// Public interface for this runtime module.
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        /// initialize the default event for this module
+        fn deposit_event() = default;
+
+        /// Reset the CAA of `ticker` to its owner.
+        ///
+        /// ## Arguments
+        /// - `ticker` for which the CAA is reset.
+        ///
+        /// ## Errors
+        /// - `BadOrigin` if not signed
+        /// - `Unauthorized` if `origin` isn't `ticker`'s owner.
+        #[weight = <T as Trait>::WeightInfo::reset_caa()]
+        pub fn reset_caa(origin, ticker: Ticker) {
+            let did = <Asset<T>>::ensure_perms_owner(origin, &ticker)?;
+            Self::change_ca_agent(did, ticker, None);
+        }
+
+        /// Set the default CA `TargetIdentities` to `targets`.
+        ///
+        /// ## Arguments
+        /// - `ticker` for which the default identities are changing.
+        /// - `targets` the default target identities for a CA.
+        ///
+        /// ## Errors
+        /// - `BadOrigin` if not signed
+        /// - `UnauthorizedAsAgent` if `origin` is not `ticker`'s sole CAA (owner is not necessarily the CAA).
+        #[weight = <T as Trait>::WeightInfo::set_default_targets(&targets)]
+        pub fn set_default_targets(origin, ticker: Ticker, targets: TargetIdentities) {
+            let caa = Self::ensure_ca_agent(origin, ticker)?;
+
+            // Dedup any DIDs in `targets` to optimize iteration later.
+            let new = {
+                let mut ts = targets;
+                ts.identities.sort_unstable();
+                ts.identities.dedup();
+                ts
+            };
+
+            // Commit + emit event.
+            let old = DefaultTargetIdentities::mutate(ticker, |slot| mem::replace(slot, new.clone()));
+            Self::deposit_event(Event::DefaultTargetIdentitiesChanged(caa, ticker, old, new));
```

Comment:
Let's skip emitting `old` in the event. It adds no useful information but increases the computation cost and storage cost.